lib/ansible/plugins/__init__.py

```diff
@@ -57,6 +59,17 @@ class AnsiblePlugin(ABC):
 
     def __init__(self):
         self._options = {}
+        self._hash = None
+
+    def _gen_signature(self):
+        # create immutable
+        s_options = pickle.dumps(self.get_options(hostvars=variables))
```

Comment:
If I copy an inventory config file and use both, should the plugin have the same hash, or be unique per source? The latter happens now with `pickle.dumps`.

Reply:
would need the same as connections it's own _gen_signature to include 'source path'

Reply:
Yeah, that would make sense. It seems like it includes source path now (implicitly) since the hashes are different, but I'm not sure exactly what's happening.